### PR TITLE
pkg/datapath/linux: Fix asymmetric IPsec logic on delete

### PR DESCRIPTION
**Pull request by christarazi**
First 3 commits are cleanups, the last commit is the main fix.

-  pkg/datapath, pkg/node/manager: Clarify NodeValidateImplementation godoc
- pkg/datapath/linux: Remove unnecessary branch in IPsec route functions
- pkg/datapath/linux: Add CIDR logfield to IPsec route logs
- pkg/datapath/linux: Fix asymmetric IPsec logic on delete

```release-note
Fix connectivity outage periods with ENI IPAM mode and IPsec enabled when nodes are deleted from the cluster
```

### CODE REVIEW

File: pkg/node/manager/manager.go [hunk 1]
```diff
@@ -307,6 +307,8 @@ func (m *Manager) backgroundSyncInterval() time.Duration {
 	return m.ClusterSizeDependantInterval(baseBackgroundSyncInterval)
 }
 
+// backgroundSync ensures that local node has a valid datapath in-place for
```
Comment (ldelossa) [inline pkg/node/manager/manager.go:310]: Stumbling on the "for every other managed node" clause here, is that worded correctly?

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -1326,10 +1326,6 @@ func (n *linuxNodeHandler) createNodeExternalIPSecOutRoute(ip *net.IPNet, dflt b
 // replaceNodeIPSecOutRoute replace the out IPSec route in the host routing table
 // with the new route. If no route exists the route is installed on the host.
 func (n *linuxNodeHandler) replaceNodeIPSecOutRoute(ip *net.IPNet) {
```
Comment (ldelossa) [inline pkg/datapath/linux/node.go:1331]: In situations like these, I like to put a small comment in the function's prose saying "leading code ensures ip will not be nil" just so later in time, no one comes in and says "oh we need to be cautious here".

**Review by ldelossa — APPROVED** [review]
LGTM, and thanks for detailed commit message. Just a few nits.

**Review by jrfastab — APPROVED** [review]

File: pkg/datapath/linux/node.go [hunk 3]
```diff
@@ -1338,7 +1338,7 @@ func (n *linuxNodeHandler) replaceNodeIPSecOutRoute(ip *net.IPNet) {
 
 	_, err := route.Upsert(n.createNodeIPSecOutRoute(ip))
```
Comment (pchaigno) [inline pkg/datapath/linux/node.go:1342]: Missing word in the description of this commit (`pkg/datapath/linux: Add CIDR logfield to IPsec route logs`):
```diff
- but we are the actual CIDR from the route that failed to be deleted or created.
+ but we are missing the actual CIDR from the route that failed to be deleted or created.
```

**Review by pchaigno — APPROVED** [review]
Only a nit below, not worth pushing if it's the only change.

**Comment by pchaigno** [comment]
k8s-1.23-kernel-net-next is failing with known flake https://github.com/cilium/cilium/issues/18447. There's no way the K8sVerifier test would be impacted by these changes since there are no bpf/ changes. So we should be ready to merge.

File: pkg/datapath/linux/node.go (hunk 3 above)
Comment (christarazi) [reply pkg/datapath/linux/node.go:1342]: Fixed.

File: pkg/node/manager/manager.go (hunk 1 above)
Comment (christarazi) [reply pkg/node/manager/manager.go:310]: Fixed.

File: pkg/datapath/linux/node.go (hunk 2 above)
Comment (christarazi) [reply pkg/datapath/linux/node.go:1331]: Fixed.

**Pull request merged**